src/Npgsql/NpgsqlConnector.cs

```diff
@@ -451,11 +450,12 @@ void WriteStartupMessage(string username)
                     "UTF8"
             };
 
-            Debug.Assert(Settings.Database != null);
-            startupParams["database"] = Settings.Database;
-            if (!string.IsNullOrEmpty(Settings.ApplicationName))
+            startupParams["database"] = Settings.Database!;
+
+            if (Settings.ApplicationName?.Length > 0)
```

Comment:
I understand this is because IsNullOrEmpty isn't recognized in older TFMs - but now I'm actually thinking we could disable nullability analysis for the older TFMs... It might make things much better. But then we'd have to suppress warnings about nullable types being used without nullability being on - let's revisit this later (have added a note to #2304)

Reply:
It isn't related to target frameworks. See dotnet/roslyn#37995. By the way, I don't like this static method, it isn't fluent and clean as this expression.

Reply:
Oh, surprising...!